src/controllers/users.go

```diff
@@ -19,13 +21,32 @@ func NewUsersController(db *gorm.DB) *UsersController {
 
 // Create new user
 func (c *UsersController) Create(email string, password string) error {
+	// Check if email is already registered
+	existingUser := &models.User{}
+	result := c.db.Where("email = ?", email).First(existingUser)
+	if result.Error == nil {
+		return errors.New("email already registered")
+	}
+	if !errors.Is(result.Error, gorm.ErrRecordNotFound) {
+		return result.Error
+	}
+
+	// Hash the password
+	hashedPassword, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	if err != nil {
```

Comment:
we can do lo.Must here, an error on hashing password is an assert level error. should crash the server

Reply:
Fair enough.